osbs/api.py

```diff
@@ -190,7 +190,7 @@ def get_build_request(self, build_type=None, inner_template=None,
         if build_type is not None:
             warnings.warn("build types are deprecated, do not use the build_type argument")
 
-        if arrangement_version < REACTOR_CONFIG_ARRANGEMENT_VERSION:
+        if not arrangement_version or arrangement_version < REACTOR_CONFIG_ARRANGEMENT_VERSION:
```

Comment:
I'd prefer if we either
* don't supply that keyword argument in `_do_create_prod_build` unless we really have a value for it (i.e. pass a **kwargs), or
* have the default value as `None` and then set it to the default as needed in this method body, e.g.
```
if arrangement_version is None:
    arrangement_version = DEFAULT_ARRANGEMENT_VERSION
```

Reply:
I wrote it this way because there are dozens of existing tests that don't pass an arrangement_version and will fail if arrangement_version == REACTOR_CONFIG_ARRANGEMENT_VERSION because they don't pass a build_type in the kwargs to build_request.set_params(), but BuildRequestV2 requires one.

I'd rather not rewrite the tests. And I can't edit the kwargs that will be used inside of build_request.set_params from inside get_build_request, so I can't go back and add a build_type if I'm returning a BuildRequestV2.

If there's a better way to do this, I'm interested, but final_version = arrangement_version or DEFAULT_ARRANGEMENT_VERSION is not right unless I rewrite all those tests.